### PR TITLE
[test] Add `@also_with_proxying` decorator. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: test/common.py [hunk 1]
```diff
@@ -2302,17 +2303,24 @@ def btest_exit(self, filename, assert_returncode=0, *args, **kwargs):
...
             reporting=Reporting.FULL,
             output_basename='test'):
     assert expected or reference, 'a btest must either expect an output, or have a reference image'
     if args is None:
       args = []
-    original_args = args
     args = args.copy()
     filename = find_browser_test_file(filename)
+
+    # Run via --proxy-to-worker.  This get set by the @also_with_proxyingZZca
```
Comment (kripken) [inline test/common.py:2316]:
```suggestion
    # Run via --proxy-to-worker.  This gets set by @also_with_proxying.
```

File: test/test_browser.py [hunk 2]
```diff
@@ -3069,13 +3097,14 @@ def test_sdl2_threads(self):
...
-    '': ([], True),
+    '': ([],),
     # fails on proxy
-    'closure': (['--closure=1', '-g1'], False),
+    'closure': (['--closure=1', '-g1'],),
   })
-  def test_sdl2_glshader(self, args, also_proxied):
-    self.reftest('test_sdl2_glshader.c', 'test_sdl_glshader.png', args=args + ['-sUSE_SDL=2', '-O2', '-sLEGACY_GL_EMULATION'], also_proxied=also_proxied)
+  def test_sdl2_glshader(self, args):
+    self.reftest('test_sdl2_glshader.c', 'test_sdl_glshader.png', args=args + ['-sUSE_SDL=2', '-O2', '-sLEGACY_GL_EMULATION'])
```
Comment (kripken) [inline test/test_browser.py:3107]: This test will double in execution time. Do you have a sense if it's worth it?

**Review by kripken — APPROVED** [review]

File: test/test_browser.py (hunk 2 above)
Comment (sbc100) [reply test/test_browser.py:3107]: Not sure, I'm happy to revert back.

Comment (sbc100) [reply test/test_browser.py]: Actually as the comment above states, closure doesn't work with proxying (or at least it generates warnings)

**Review by kripken — APPROVED** [review]

**Pull request merged**